automatic/nexus-repository/tools/chocolateyinstall.ps1

```diff
@@ -47,6 +48,76 @@ if ($pp.ContainsKey("BackupSslConfig")) {
   }
 }
 
+if ($CurrentlyInstalledVersion -and [version]($CurrentlyInstalledVersion -replace '-\d+$') -lt "3.70") {
+  Write-Error (@(
+    "Please upgrade nexus-repository to version 3.70.1-02 before upgrading further."
+    "You can do this by running 'choco upgrade nexus-repository --version 3.70.1.2 --confirm'"
+    "For more details, please see: https://help.sonatype.com/en/orient-pre-3-70-java-8-or-11.html"
+  ) -join "`n")
+  throw "Package cannot upgrade from '$($CurrentlyInstalledVersion)' to '$($env:ChocolateyPackageVersion)'"
+} elseif ($CurrentlyInstalledVersion -and [version]($CurrentlyInstalledVersion -replace '-\d+$') -eq "3.70.1" -and (Test-NexusDatabaseType -Type "OrientDb" -DataDir $TargetDataFolder -ProgramDir $TargetFolder)) {
+  Write-Host "Preparing for database migration..."
+  $MigrationFiles = Join-Path $ExtractFolder "dbmigration"
+  if ((Get-PSDrive $MigrationFiles[0]).Free -lt ($RequiredSpace = [Math]::Max((3 * (Get-ChildItem -Path $TargetDataFolder\nexus3\db -Recurse | Measure-Object Length -Sum).Sum), 10GB))) {
```

Comment:
We may want to break this down a bit so it's easier to follow what's goin on. It works, but it is a bit of a symbol salad right now.

Reply:
I did think about pushing this all out into a Test-migrationrequired function. It would be moving the problem, but maybe good for those who read the installer scripts!

Reply:
Someone still has to review the code though whether it is here directly or in helper. Pick a spot, and use more variable for readability. We don't have to pay for them :)

Reply:
Ah, I misunderstood your complaint.